core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/component/dto/ComponentRepositoryDtoTest.java

```diff
@@ -0,0 +1,250 @@
+import io.metadew.iesi.metadata.definition.component.ComponentVersion;
+import io.metadew.iesi.metadata.definition.component.key.ComponentKey;
+import io.metadew.iesi.metadata.definition.component.key.ComponentParameterKey;
+import io.metadew.iesi.metadata.definition.component.key.ComponentVersionKey;
+import io.metadew.iesi.server.rest.Application;
+import io.metadew.iesi.server.rest.component.ComponentFilter;
+import io.metadew.iesi.server.rest.component.ComponentFilterOption;
+import io.metadew.iesi.server.rest.configuration.TestConfiguration;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.ArrayList;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = Application.class, properties = {"spring.main.allow-bean-definition-overriding=true"})
+@ContextConfiguration(classes = TestConfiguration.class)
+@ActiveProfiles("test")
+@DirtiesContext
+public class ComponentRepositoryDtoTest {
+
+    @Autowired
+    private ComponentDtoRepository componentDtoRepository;
+    @Autowired
+    private ComponentConfiguration componentConfiguration;
+    @Autowired
+    private MetadataRepositoryConfiguration metadataRepositoryConfiguration;
+    @Autowired
+    private ComponentDtoResourceAssembler componentDtoResourceAssembler;
+
+    @AfterEach
+    void cleanup() {
+        metadataRepositoryConfiguration.clearAllTables();
+    }
+
+    @Test
+    void getAllNoResultsTest() {
+        assertThat(componentDtoRepository.getAll(Pageable.unpaged(), new ArrayList<>()))
+                .isEmpty();
+    }
+
+    @Test
+    void getAllTest() {
+        Component component1 = createComponent("component1");
+        Component component2 = createComponent("component2");
+        componentConfiguration.insert(component1);
+        componentConfiguration.insert(component2);
+        ComponentDto component1Dto = componentDtoResourceAssembler.toModel(component1);
+        ComponentDto component2Dto = componentDtoResourceAssembler.toModel(component2);
+        assertThat(componentDtoRepository.getAll(Pageable.unpaged(), new ArrayList<>())).containsOnly(
+                component1Dto,
+                component2Dto
+        );
+    }
+
+    @Test
+    void getAllPaginatedAllInclusiveTest() {
```

Comment:
also test 1 case where there are 3 components in the DB and you fetch a page of 2.
Can you also check the page size being returned?

Reply:
The repository returns the component in any order and the tests is randomly passed.
Sometime the page 1 can return ;
- Component1 and Component3
- Component2 and Component3
-  ...

The issue is that the order by clause is not performant. There are no way to select component in order of when they are created, the database returns random components. 

The repository order by ID if there are not filters,but the ids cannot by ordered because they are randomly generated due to the lack of ordering pattern ( from 0-9, from A-Z, from date, ... ) : 

- 3f662dd1-b9ea-4890-95b8-90f385dd88cf
- eae14c4e-52ce-4d4c-bf83-04cde79b9384
- ...

Maybe is it possible to create new column such as "createdAt" to trace the component age ?